### ISSUE TITLE
Streamline notification settings

### ISSUE THREAD

**Issue by xPaw** [thelounge/lounge#1441]
2.5.0 adds push notifications, and we want to get the UI and UX right for the settings before we release.

As we discussed with @astorije, we decided that 3 (or 4 depending on how you look) is the best way to go here, this will accomplish different workflows that users want, and account for platform differences.

- No notifications
- Notifications only when the app is open
- Notifications only when the app is closed
- Always notify (highlights that is)

We also think that "notification for all messages" is too niche of an option and should be removed.

**Comment by YaManicKill** [comment]
So, for a default, number 3 definitely makes sense. 

However (and I realise this won't work just now because we need to get settings sync sorted out) it would be better if we have these as options for the user, because these are entirely subjective settings.

**Comment by xPaw** [comment]
Default would be no notifications. Do keep in mind that notifications require a permission that the user has to enable in the browser.

**Comment by YaManicKill** [comment]
Obviously, I meant default after enabling notifications.

**Comment by xPaw** [comment]
I think the idea is to make these checkboxes/radio, and it would ask for permissions when you select the option you want.

**Comment by YaManicKill** [comment]
Oh right, so on a client-per-client basis? That would make sense.

**Comment by YaManicKill** [comment]
I guess this doesn't solve the "client open on another machine" problem, then?

**Comment by xPaw** [comment]
> I guess this doesn't solve the "client open on another machine" problem, then?

Server already pushes to clients (sessions) that are not online. And we will only need to register web push with the server if client wants to get offline notifications.

**Comment by YaManicKill** [comment]
> Server already pushes to clients (sessions) that are not online. And we will only need to register web push with the server if client wants to get offline notifications.

I'm obviously not explaining myself properly.

Current situation:

1. I have a desktop and a phone
2. Phone is registered for push notifications
3. Lounge is closed on phone
4. Lounge is open on desktop
5. When I get a highlight, I get a notification on phone

Ideal situation:

1. I have a desktop and a phone
2. Phone is registered for push notifications
3. Lounge is closed on phone
4. Lounge is open on desktop
5. When I get a highlight, I only get notification on desktop
6. Lounge is closed on phone and desktop
7. When I get a highlight, I get a notification on phone

Does that make sense? As I mentioned, this will need syncing settings to server, but it's do-able, and would be a nice feature.

**Comment by xPaw** [comment]
@YaManicKill It worked this way before, when any client was attached, it would simply not send any push notifications. There's certainly different use cases that different users want here.

**Comment by YaManicKill** [comment]
> There's certainly different use cases that different users want here.

Yeah, I'm not asking for this to be a default, just that it can be done at some point. What I was saying was I originally thought this ticket was for this feature, but it isn't, is it?

**Comment by YaManicKill** [comment]
I know this isn't fully complete from #1468, but we aren't going to put any more PRs into 2.5.0 from this, so I'll take the milestone off it.

**Comment by astorije** [comment]
Eh, I guess. @xPaw, what do you think?

**Comment by xPaw** [comment]
I'm fine with that.

**Comment by nsrafk** [comment]
> > Server already pushes to clients (sessions) that are not online. And we will only need to register web push with the server if client wants to get offline notifications.
> 
> 
> 
> I'm obviously not explaining myself properly.
> 
> 
> 
> Current situation:
> 
> 
> 
> 1. I have a desktop and a phone
> 
> 2. Phone is registered for push notifications
> 
> 3. Lounge is closed on phone
> 
> 4. Lounge is open on desktop
> 
> 5. When I get a highlight, I get a notification on phone
> 
> 
> 
> Ideal situation:
> 
> 
> 
> 1. I have a desktop and a phone
> 
> 2. Phone is registered for push notifications
> 
> 3. Lounge is closed on phone
> 
> 4. Lounge is open on desktop
> 
> 5. When I get a highlight, I only get notification on desktop
> 
> 6. Lounge is closed on phone and desktop
> 
> 7. When I get a highlight, I get a notification on phone
> 
> 
> 
> Does that make sense? As I mentioned, this will need syncing settings to server, but it's do-able, and would be a nice feature. 

I very much agree here. Being able to set settings to get the ideal setup would be epic. Its annoying to receive push notifications on phone when chatting with ppl on the desktop..
